src/firebase/afterDragAndDrop.js

```diff
@@ -0,0 +1,97 @@
+import {
+  collection,
+  deleteDoc,
+  doc,
+  getDoc,
+  getDocs,
+  setDoc,
+} from "firebase/firestore";
+
+import { db } from "./firebase";
+
+async function getHistoryIdMapByDB(userId) {
+  const historyIdMap = new Map();
+
+  const groupIds = [];
+
+  const groupsRef = collection(db, "users", userId, "groups");
+  const groupsQuerySnapshot = await getDocs(groupsRef);
+  groupsQuerySnapshot.forEach((groupDoc) => {
+    groupIds.push(groupDoc.id);
+  });
+
+  for (const groupId of groupIds) {
+    const historiesRef = collection(
+      db,
+      "users",
+      userId,
+      "groups",
+      groupId,
+      "histories"
+    );
+
+    const historiesQuerySnapshot = await getDocs(historiesRef);
+    historiesQuerySnapshot.forEach((historyDoc) => {
+      historyIdMap.set(historyDoc.id, groupId);
+    });
+  }
+
+  return historyIdMap;
+}
+
+export async function updateGroupsAndHistoriesAfterDragAndDrop(
+  userId,
+  updatedHistoryGroups
+) {
+  const originHistoryIdMap = await getHistoryIdMapByDB(userId);
+
+  for (let i = 0; i < updatedHistoryGroups.length; i += 1) {
+    const updatedGroup = updatedHistoryGroups[i];
+    const updatedGroupId = updatedGroup.id;
+    const historiesByUpdatedGroup = updatedGroup.histories;
+
+    const historiesRef = collection(
+      db,
+      "users",
+      userId,
+      "groups",
+      updatedGroupId,
+      "histories"
+    );
+
+    for (const history of historiesByUpdatedGroup) {
+      const historyId = history.id;
+
+      const historyDocRef = doc(
+        db,
+        "users",
+        userId,
+        "groups",
+        updatedGroupId,
+        "histories",
+        historyId
+      );
+      const historyDocSnap = await getDoc(historyDocRef);
+
+      if (!historyDocSnap.exists()) {
```

Comment:
p4: "historyDocSnap에 data가 없다면"을 표현하고 싶으셨던게 맞으실까요?
제가 해석한 바가 맞다면 NOT연산자 보단 동등연산자를 사용하여 가독성을 높여보는 것은 어떨까요?
```suggestion
      if (historyDocSnap.exists() === false) {
```

Reply:
네 맞습니다. "해당 id를 가진 history가 해당 위치에 없다면"을 표현했습니다. 참고하기로 한 airbnb 스타일 가이드에서 다음과 같이 단축형을 쓸 것을 얘기하는 부분이 있었습니다. 이에 현재 가독성도 괜찮아 보이는데, 변경하는 것도 좋습니다. P4인 만큼 팀 미팅때 간단하게 논의를 해보고 싶은데 어떠신가요?

https://github.com/tipjs/javascript-style-guide?tab=readme-ov-file#15.3

Reply:
말씀주신대로 취향적인 부분이 있는 것 같습니다. 저도 airbnb에는 단축형을 권한다는 것을 알고는 있지만, not연산자가 쓰일 때는 동등연산자로 좀 더 직관적으로 보이면 어떨까 싶어서 의견드려봤습니다. 이 부분은 회의를 통해 논의해보면 좋을 것 같습니다.